enabler/src/de/schildbach/pte/dto/Trip.java

```diff
@@ -150,6 +150,35 @@ public Date getMaxTime() {
         return maxTime;
     }
 
+    /** Number of Changes on the trip.
+     *
+     * Returns the variable numChanges if it is not null.
+     * If numChanges is null, it tries to reconstruct the number of changes where applicable.
+     * The number of changes for a Trip consisting of only individual Legs is null.
+     * @author Patrick Kanzler
+     *
+     * @return Number of changes on the trip
+     */
+    @Nullable
+    public Integer getNumChanges() {
+        if (numChanges == null) {
+            Integer numCount = null;
+
+            for (final Leg leg : legs){
+                if (!(leg instanceof Individual)) {
```

Comment:
I mistook `Leg` for being instatiable. So I tried to test for the case that we don't have an `Individual`. I will change this to a test for `Public`, because that is more explicit.